lib/computer.rb

```diff
@@ -1,15 +1,50 @@
-require "./spec/spec_helper"
-
 class Computer
-  attr_reader :mark
+  attr_reader :mark, :opponent
 
-  def initialize(mark)
-    @moves = computer_player
+  def initialize(mark, board)
     @mark = mark
+    @board = board
   end
 
   def get_move
-    @moves.shift
-    # return a random available move
+    analyse_board
+  end
+
+  def opponent_mark
+    if @mark == "O"
+      opponent = "X"
+    else 
+      opponent = "O"
+    end
+    opponent
+  end
+
+  def analyse_board
+    spaces = @board.available_moves
+    current_board = @board.board_array
+    best_play = nil 
+
+    spaces.each do |space|
+      @board.mark_board(space, @mark)
+      if @board.current_player_wins?(@mark)
```

Comment:
I find the `current_player_wins?` method name confusing, given that it expects a player's mark as an argument. Could you rename this method (I assume it's checking the player with the provided mark has won)?

Reply:
Resolved #30